pkg/types/validation/installconfig.go

```diff
@@ -54,7 +54,7 @@ import (
 var pluginsUsingHostPrefix = sets.NewString(string(operv1.NetworkTypeOpenShiftSDN), string(operv1.NetworkTypeOVNKubernetes))
 
 // ValidateInstallConfig checks that the specified install config is valid.
-func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
+func ValidateInstallConfig(c *types.InstallConfig, agentMethod bool) field.ErrorList {
```

Comment:
I understand that this boolean has been kept to maintain the same previous behavior (cherry-picking the specific validations to enable/disable). This refactoring could open the possibility in the future to change approach and prepare the installconfig accordingly before the validation step.

nit: `agentMethod` is a little bit confusing, since it's not a method... I'd suggest to keep the previous `agentBasedInstallation`